src/DeterministicNoiseSimulator.cpp

```diff
@@ -2,73 +2,87 @@
+////        assert(true); // This function does not work yet.
+//////        dEdge newEdge{dd->dUniqueTable.getNode(), e.w};
+//////        auto tmop = dEdge::getAlignedDensityNodeCopy(e, &newEdge);
+////    }
+//
+//    if (e.w.approximatelyZero() || (e.p->v == index && e.p->e[0].w.approximatelyZero())) {
+//        return 0;
+//    }
+//
+//    assert(!e.isTerminal());
+//    assert(dd::CTEntry::val(e.w.i) == 0);
+//    pathProb = pathProb * dd::CTEntry::val(e.w.r);
+//
+//    if (e.p->v > index) {
+//        global_prob += probForIndexToBeZero(e.p->e[0], index, pathProb, global_prob);
+//        global_prob += probForIndexToBeZero(e.p->e[3], index, pathProb, global_prob);
+//        return global_prob;
+//    } else {
+//        return pathProb;
+//    }
+//}
+
+//char DeterministicNoiseSimulator::MeasureOneCollapsing(dd::Qubit index) {
+//    char                                   result   = 'n';
+//    auto                                   f        = qc::MatrixDD::one;
+//    auto                                   n_qubits = getNumberOfQubits();
+//    std::uniform_real_distribution<dd::fp> dist(0.0, 1.0L);
+//    dd::fp                                 n = dist(mt);
+//
+//    auto prob_zero = probForIndexToBeZero(density_root_edge, index, 1, 0);
+//
+//    for (dd::Qubit p = 0; p < (signed char)n_qubits; p++) {
+//        if (p == index) {
+//            if (prob_zero >= n) {
+//                // Build the operation that it sets index to 0
+//                result = '0';
+//                f      = dd->makeDDNode(static_cast<dd::Qubit>(p), std::array{f, qc::MatrixDD::zero, qc::MatrixDD::zero, qc::MatrixDD::zero});
+//            } else {
+//                result = '1';
+//                // Build the operation that it sets index to 1
+//                f = dd->makeDDNode(static_cast<dd::Qubit>(p), std::array{qc::MatrixDD::zero, qc::MatrixDD::zero, qc::MatrixDD::zero, f});
+//            }
+//        } else {
+//            f = dd->makeDDNode(static_cast<dd::Qubit>(p), std::array{f, qc::MatrixDD::zero, qc::MatrixDD::zero, f});
+//        }
+//    }
+//    //    qc::MatrixDD tmp0 = dd->multiply(dd->multiply(f, density_root_edge), dd->conjugateTranspose(f));
+//    auto tmp0 = dd->conjugateTranspose(f);
+//    auto tmp1 = dd->multiply(density_root_edge, reinterpret_cast<dEdge&>(tmp0), 0, false);
+//    auto tmp2 = dd->multiply(reinterpret_cast<dEdge&>(f), tmp1, 0, true);
+//
+//    dd->incRef(tmp2);
+//    dd->decRef(density_root_edge);
+//    density_root_edge = reinterpret_cast<dEdge&>(tmp2);
+//
+//    // Normalize the density matrix
+//    auto trace = dd->trace(reinterpret_cast<dd::mEdge&>(density_root_edge));
+//    assert(trace.i == 0);
+//    auto tmp_complex = dd->cn.getTemporary(1 / trace.r, 0);
+//    dd::ComplexNumbers::mul(tmp_complex, density_root_edge.w, tmp_complex);
+//    dd::ComplexNumbers::decRef(density_root_edge.w);
+//    density_root_edge.w = dd->cn.lookup(tmp_complex);
+//    return result;
+//}
```

Comment:
```suggestion

```

Reply:
Removed the code and added an issue to reimplement the functions